CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationRewards.uc

```diff
@@ -37,6 +40,110 @@ static function X2DataTemplate CreateDatapadRewardTemplate()
 	return Template;
 }
 
+static function X2DataTemplate CreateContainerRewardTemplate()
+{
+	local X2RewardTemplate Template;
+
+	`CREATE_X2Reward_TEMPLATE(Template, 'Reward_Container');
+
+	Template.GenerateRewardFn = GenerateContainerReward;
+	Template.SetRewardFn = SetContainerReward;
+	Template.GiveRewardFn = GiveContainerReward;
+	/*
+	Template.GetRewardStringFn = class'X2StrategyElement_DefaultRewards'.static.GetItemRewardString;
+	Template.GetRewardImageFn = class'X2StrategyElement_DefaultRewards'.static.GetItemRewardImage;
+	Template.GetBlackMarketStringFn = class'X2StrategyElement_DefaultRewards'.static.GetItemBlackMarketString;
+	Template.GetRewardIconFn = class'X2StrategyElement_DefaultRewards'.static.GetGenericRewardIcon;
+	Template.RewardPopupFn = class'X2StrategyElement_DefaultRewards'.static.ItemRewardPopup;
+	*/
+	return Template;
+}
+
+static function GenerateContainerReward(XComGameState_Reward RewardState, XComGameState NewGameState, optional float RewardScalar = 1.0, optional StateObjectReference RegionRef)
```

Comment:
This seems a really complicated way of doing this. I would get rid of the `GenerateRewardFn` delegate/function entirely and call set reward inside chain's `PostStageSetup`

Reply:
Delete the function entirely, not just comment it out. You always have the git history anyway